tpm/storage/types.go

```diff
@@ -0,0 +1,152 @@
+package storage
+
+import (
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+type AK struct {
+	Name      string
+	Data      []byte
+	Chain     []*x509.Certificate
+	CreatedAt time.Time
+}
+
+func (ak *AK) MarshalJSON() ([]byte, error) {
+	chain := make([][]byte, len(ak.Chain))
+	for i, cert := range ak.Chain {
+		chain[i] = cert.Raw
+	}
+
+	sak := serializedAK{
+		Name:      ak.Name,
+		Type:      typeAK,
+		Data:      ak.Data,
+		CreatedAt: ak.CreatedAt,
+	}
+
+	if len(chain) > 0 {
+		sak.Chain = chain
+	}
+
+	return json.Marshal(sak)
+}
+
+func (ak *AK) UnmarshalJSON(data []byte) error {
+	sak := &serializedAK{}
+	if err := json.Unmarshal(data, sak); err != nil {
+		return fmt.Errorf("failed unmarshaling serialized AK: %w", err)
+	}
+
+	ak.Name = sak.Name
+	ak.Data = sak.Data
+	ak.CreatedAt = sak.CreatedAt
```

Comment:
Do we need to verify the Type == typeAK?

Reply:
Nice one. Added in 1d8a8b434257ae7d1338ba30b54f9bf8abb93ca4.